### ISSUE TITLE
Move configuration from global mutable variable to Reader context or at least pass explicitly

### ISSUE THREAD

**Issue by chshersh** [serokell/log-warper#47]
Currently all IO configuration is done using global `MVar` with `unsafePerformIO`. This is not really good :(

**Comment by chshersh** [comment]
Here I will describe plan to migrate to newer interface.

### What we have now:

#### We can ask `LoggerName`
```haskell
newtype LoggerNameBox m a = LoggerNameBox { loggerNameBoxEntry :: ReaderT LoggerName m a }

class HasLoggerName m where
    askLoggerName    :: m LoggerName
    modifyLoggerName :: (LoggerName -> LoggerName) -> m a -> m a
```

#### We can dispatch messages

```haskell
class Monad m => CanLog m where
    dispatchMessage :: LoggerName -> Severity -> Text -> m ()

instance CanLog IO where dispatchMessage = logM
```

#### Global configuration stored in global mutable variable

```haskell
{-# NOINLINE logInternalState #-}
logInternalState :: MVar LogInternalState
-- note: only kick up tree if handled locally
logInternalState = unsafePerformIO $ do ...
```

### What I propose

We probably want to switch to some another approach, for example: [`caps`](https://github.com/int-index/caps/). But at least `LogInternalState` should be stored in `ReaderT` context. Things become more complex because we want to support pure logging as well and we don't need `LogInternalState` there.

So let's have:

```haskell
newtype LoggerContext = ... -- old `LogInternalState`

newtype LoggerT m a = LoggerT { runLoggerT :: ReaderT (TVar LoggerContext) m a }
```

This will make transition more smooth. We are not going to remove all `IO` functions, they just will work inside `LoggerT` monad.

Here is the plan to do this:

1. Rename `LogInternalState` to `LoggerContext`.
2. Add `LoggerT` newtype with all required derivings somewhere in its own module with `LogInternalState`.
3. Refactor all needed functions inside `IOLogger` to user `LoggerT` type.
4. `IO` can no longer have `CanLog` instance. Instead `LoggerT` should have this instance implemented.
5. Remove `buildAndSetupYamlLogging`.
6. Now, the interesting and most difficult part: `setupLogging` function cannot longer exist in a way it exists now. The only ways to run logging should be the following:
    + `launchWithConfig`
    + `launchFromFile`
    + `launchSimpleLogging`
  Inside these functions we want to:
    + Create `TVar` with empty internal state which [is here now](https://github.com/serokell/log-warper/blob/efd07d58d2f0a2bfe5c7c2d191f4776beb47da37/src/System/Wlog/IOLogger.hs#L110) (`logInternalState` variable should be removed after that)
    + Run `setupLogging` which updates this `TVar`
    + Clears this `TVar` after everything is finished
    + `bracket` still should be used

Probably some technical details will arise during implementation..

**Comment by chshersh** [comment]
Fixed in version 2.0.
